app/main/views/auth.py

```diff
@@ -131,6 +139,12 @@ def authentication_request():
         session['suggested_idp'] = request.args['suggested_idp']
         return redirect(url_for('.confirm_idp'))
 
+    elif request.cookies.get('gateway_idp'):
+        session['suggested_idp'] = request.cookies.get('gateway_idp')
+        department = dept_from_idp_name(session['suggested_idp'])
+        session['department_name'] = department
```

Comment:
The `department` variable is only used once immediately after it is set, so it should probably be inlined.
```python
session['department_name'] = dept_from_idp_name(session['suggested_idp'])
```